spec/models/event_stream_spec.rb

```diff
@@ -0,0 +1,28 @@
+def with_event_types
+  EventStream.event_groups.each do |group_name, group_data|
+    EventStream::GROUP_LEVELS.each do |level|
+      group_data[level]&.each { |typ| yield(group_name, level, typ) }
+    end
+  end
+end
```

Comment:
Prefer defining this *within* the describe block, so it doesn't pollute the global namespace.

Reply:
RSpec will not allow this. Things that are defined inside `describe` block can only be used from within the `it` block.

Another option would be to use something like [Rspec::Parameterized][rparams] or write a simple helper for this purpose.

   [rparams]: https://github.com/tomykaira/rspec-parameterized

If we do not care about the "developer friendliness", we can loop inside the `it` block and sidestep all this, but I feel that this is not the way to go.

Reply:
Does it make sense to add this as an actual method to the model?  Then, it can be possibly be used in the UI code as well?

Reply:
@bdunne Thoughts here?  I thought we did something like this elsewhere.

Reply:
We do something like [this](https://github.com/ManageIQ/manageiq/blob/master/spec/models/miq_provision_workflow_spec.rb#L101-L105) in many places if that's what you mean.  :+1: I agree with not polluting the global namespace.